File: src/foo/deep/Row.java
package foo.deep;

import java.util.HashMap;
import java.util.Map;

public class Row {

    private Map<String, Object> attributes = new HashMap<String, Object>();
    private String tableName;

    public Row(String tableName) {
        this.tableName = tableName;
    }

    public void setAttribute(String name, Object value) {
        attributes.put(name, value);
    }

    public Object getAttribute(String name) { return attributes.get(name); }

    public String getTableName() { return tableName; }

}

File: src/foo/row.rb
require 'rubygems'
require 'java'
require 'jruby/core_ext'

module F
    include_package "foo.deep"
end

include F

class Java::FooDeep::Row

    def self.inherited(subclass)

        subclass.class_eval do

            table_name = subclass.to_s

            define_method :initialize do |*args|
                super(table_name)
            end

        end

    end
end